app/src/main/java/com/svalero/appaprendizaje/listnasa/view/List_nasa_activity.java

```diff
@@ -33,25 +36,47 @@ protected void onCreate(Bundle savedInstanceState) {
         list_nasa_presenter = new List_nasa_presenter(this);
         list_nasa_presenter.getListNasa();
 
-        titulo.setOnClickListener(new View.OnClickListener() {  //Operador ternario para que el mismo botón cambie del fragment a la activity y viceversa
+        titulo.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
-                transaction.replace(R.id.fragmentContainer,fragment).commit();
+
+                FragmentManager fragmentManager = getSupportFragmentManager();
+                Fragment fragment = fragmentManager.findFragmentById(R.id.fragmentContainer);
+
+                System.out.println(fragment);
+
+                /*if(fragment instanceof MainFragment) {
+                    showFragment(DescriptionFragment.newInstance(nasa));
+                } else {
+                    showFragment(MainFragment.newInstance(nasa));
+                }*/
+
+                Fragment newFragment = fragment instanceof MainFragment ? DescriptionFragment.newInstance(nasa) : MainFragment.newInstance(nasa);
+                showFragment(newFragment);
             }
         });
     }
 
     public void initComponents() {
         titulo = findViewById(R.id.bTitle);
         imagen = findViewById(R.id.ivImage);
-        fragment = new DescriptionFragment();
+    }
+
+    public void showFragment(Fragment fragment) {
+        getSupportFragmentManager().beginTransaction().replace(R.id.fragmentContainer, fragment).commit();
+    }
+
+    @Override
+    public void onBackPressed() {
+        showFragment(MainFragment.newInstance(nasa));
     }
 
     @Override
     public void success(Nasa nasa) {
-       titulo.setText(nasa.getTitle());
-       Picasso.get().load(nasa.getUrl()).into(imagen);
+       this.nasa = nasa;
+       titulo.setText(nasa.getTitulo());
+       showFragment(MainFragment.newInstance(nasa));
+        Toast.makeText(this, "Hola !!!!!", Toast.LENGTH_SHORT).show();
```

Comment:
Toast.makeText(this, "Hola !!!!!", Toast.LENGTH_SHORT).show();
```suggestion
        Toast.makeText(this, "Hola", Toast.LENGTH_SHORT).show();
```